osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/security/providers/JreEncryptionProvider.java

```diff
@@ -72,16 +78,34 @@ public String getFormat() {
 
             @Override
             public byte[] getEncoded() {
-                return key;
+                return JreEncryptionProvider.this.key;
             }
         };
     }
 
+    @Override
     protected AlgorithmParameterSpec getAlgorithmParameterSpec() {
         return new IvParameterSpec(IV);
     }
 
+    @Override
+    public byte[] generateAes128BitsSecret(String keyReference) {
+        try {
+            KeyGenerator keyGenerator = KeyGenerator.getInstance("AES");
```

Comment:
Missing final modifier on local variable `keyGenerator` (and `exc` a few lines down).